source/developers/concepts/hooks_extension_framework.rst

```diff
@@ -156,6 +156,39 @@ Filters are useful when you need an immediate response that directly modifies th
+| **Advantages**                                                     | - More decoupled implementations.                                                                             | - Full control of the platform behavior.                                                                                           |
+|                                                                    | - Decreased maintenance burden of the platform, maintenance efforts are centered on plugins.                  | - No restrictions from what can be changed.                                                                                        |
+|                                                                    | - Enables reusable and configurable behavior across the platform.                                             | - Suitable for highly customized use cases.                                                                                        |
+|                                                                    | - Easier to maintain long-term due to backward compatibility and versioning policies.                         |                                                                                                                                    |
++--------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| **Disadvantages**                                                  | - Can increase complexity in debugging and maintenance because it adds more processing layers.                | - Separation from the core makes long-term maintenance challenging.                                                                |
+|                                                                    | - Limited by existing events and filters available in the Open edX ecosystem, although new events and         | - High maintenance burden for upgrades and compatibility.                                                                          |
+|                                                                    |   filters can be proposed and added to the framework.                                                         | - Requires manual integration with the Open edX platform for each new release.                                                     |
+|                                                                    | - Might increase complexity in plugins for highly customized features.                                        | - Requires deep understanding of the platform to modify it correctly.                                                              |
```

Comment:
> Requires deep understanding of the platform to modify it correctly

I would argue that modifying correctly means using the open-closed principle which is not being used if the road was a fork. 

I would list however the "not being able to reuse by the community" as part of the disadvantages.

Reply:
Aso a disadvantage of fork:

- difficult to test in isolation

Reply:
> I would argue that modifying correctly means using the open-closed principle which is not being used if the road was a fork.

I removed the "correctly" from the sentence. I do think that to modify the platform, you need a deep understanding of the components you'd be modifying, mainly because of the implications of the changes. But one can argue that you'd need the same to add new extension points, so I'll drop the line altogether. 

> I would list however the "not being able to reuse by the community" as part of the disadvantages.

Added! Thanks https://github.com/openedx/docs.openedx.org/pull/617/commits/67bd6f7c4060421d7b2dea42c6affffa8a8d6acd#diff-01d0bcd5d9113aaa8a7f3fde94c7e5ca100265eebebbad867dbf4de21123d451R181